src/codegen/codegen_cpp_visitor.cpp

```diff
@@ -614,18 +598,12 @@ std::vector<std::string> CodegenCVisitor::ion_read_statements(BlockType type) {
     for (const auto& ion: info.ions) {
         auto name = ion.name;
         for (const auto& var: ion.reads) {
-            if (type == BlockType::Ode && ion.is_ionic_conc(var) && state_variable(var)) {
-                continue;
-            }
             auto variable_names = read_ion_variable_name(var);
```

Comment:
hmm...considering this usage (which appears quite sensitive), I will give a detailed look tomorrow.

Reply:
Just to have this for future reference, this condition was based on the original implementation here: https://github.com/BlueBrain/mod2c/blob/284bd611a349b563a6ed8fb35a42e63e32860c0c/src/mod2c_core/nocpout.c#L2416

where block type == 2 supposed to be:

```cpp
List *get_ion_variables(block)
	int block;	/* 0 means equation block */
		    /* 2 means ode_spec and ode_matsol blocks */
```

But looking at `nrn/src/nmodl` as well as `mod2c/src` I could only find:

```console
➜  mod2c git:(master) ✗ grep -r get_ion_variables src/mod2c_core/*
src/mod2c_core/noccout.c:	printlist(get_ion_variables(1));
src/mod2c_core/noccout.c:	  printlist(get_ion_variables(0));
src/mod2c_core/noccout.c:	  printlist(get_ion_variables(1));
src/mod2c_core/noccout.c:	printlist(get_ion_variables(1));
src/mod2c_core/noccout.c:	  printlist(get_ion_variables(0));
src/mod2c_core/noccout.c:	  printlist(get_ion_variables(1));
src/mod2c_core/nocpout.c:		lst = get_ion_variables(0);
src/mod2c_core/nocpout.c:List *get_ion_variables(block)
src/mod2c_core/nocpout.c:		lst = get_ion_variables(1);
src/mod2c_core/nocpout.c:		lst = get_ion_variables(1);

➜  nmodl git:(master) ✗ grep -r get_ion_variables *
noccout.cpp:    printlist(get_ion_variables(1));
noccout.cpp:            printlist(get_ion_variables(0));
noccout.cpp:        printlist(get_ion_variables(1));
noccout.cpp:    printlist(get_ion_variables(1));
noccout.cpp:            printlist(get_ion_variables(0));
noccout.cpp:        printlist(get_ion_variables(1));
nocpout.cpp:            lst = get_ion_variables(0);
nocpout.cpp:            lst = get_ion_variables(1);
nocpout.cpp:            lst = get_ion_variables(1);
```

So this is good to ignore!